ci/release/apply_wheel_modifications.sh

```diff
@@ -9,3 +9,7 @@ CUDA_SUFFIX=${2}
 sed -i "s/^version = .*/version = \"${VERSION}\"/g" python/pyproject.toml
 
 sed -i "s/^name = \"rmm\"/name = \"rmm${CUDA_SUFFIX}\"/g" python/pyproject.toml
+
+if [[ $CUDA_SUFFIX == "-cu12" ]]; then
+    sed -i "s/cuda-python.*/cuda-python>=12.0,<13.0\",/g" python/pyproject.toml
```

Comment:
Avoid replacing quotes/commas in the syntax. Just replace the version specifier. (I did test this one.)
```suggestion
    sed -i "s/cuda-python[<=>\.,0-9]*/cuda-python>=12.0,<13.0/g" python/pyproject.toml
```